sendgrid_test.go

```diff
@@ -1852,12 +1852,40 @@ func Test_test_mail_batch__batch_id__get(t *testing.T) {
 	assert.Equal(t, 200, response.StatusCode, "Wrong status code returned")
 }
 
+type U struct{}
+
+func (u U) templateExists(c Client, ID string) (*rest.Response, error) {
+	res := &rest.Response{StatusCode: 404}
+	err := &rest.RestError{Response: res}
+	return res, err
+}
+
+func Test_test_send_client_with_invalid_template(t *testing.T) {
+	apiKey := "SENDGRID_APIKEY"
+	client := NewSendClient(apiKey)
+	client.Validator = &U{}
+	m := &mail.SGMailV3{}
+	m.SetTemplateID("[YOUR TEMPLATE ID GOES HERE]")
+	response, err := client.Send(m)
+	if err != nil {
+		fmt.Println(err)
+	}
+	assert.Equal(t, 404, response.StatusCode, "Wrong status code returned")
+}
+
+type V struct{}
```

Comment:
consider renaming `V` to something like `validTemplateRequest`

Reply:
Thanks for your suggestion @dmowcomber.
I'll make these changes in a separate PR.